app/src/main/java/org/fossasia/openevent/general/utils/extensions/ViewExt.kt

```diff
@@ -24,3 +28,110 @@ fun View.showWithFading(duration: Long = 200L) {
+        val matchParentMeasureSpec = View.MeasureSpec.makeMeasureSpec(viewParent.width, View.MeasureSpec.EXACTLY)
+        val wrapContentMeasureSpec = View.MeasureSpec.makeMeasureSpec(0, View.MeasureSpec.UNSPECIFIED)
+        measure(matchParentMeasureSpec, wrapContentMeasureSpec)
+    }
+    val targetHeight = measuredHeight
+
+    layoutParams.height = 1
+    isVisible = true
+    val animation = object : Animation() {
+        override fun applyTransformation(interpolatedTime: Float, t: Transformation?) {
+            layoutParams.height = if (interpolatedTime == 1F) ViewGroup.LayoutParams.WRAP_CONTENT
+                            else (targetHeight * interpolatedTime).toInt()
+            requestLayout()
+        }
+
+        override fun willChangeBounds(): Boolean {
+            return true
+        }
+    }
+    animation.duration = 300
+    startAnimation(animation)
+}
+
+fun TextView.expandTextView() {
+    val currentHeight = measuredHeight
+    minLines = 0
+    maxLines = Int.MAX_VALUE
+    val viewParent = parent
+    if (viewParent is View) {
+        val matchParentMeasureSpec = View.MeasureSpec.makeMeasureSpec(viewParent.width, View.MeasureSpec.EXACTLY)
+        val wrapContentMeasureSpec = View.MeasureSpec.makeMeasureSpec(0, View.MeasureSpec.UNSPECIFIED)
+        measure(matchParentMeasureSpec, wrapContentMeasureSpec)
+    }
+    val targetHeight = measuredHeight
+
+    layoutParams.height = 1
+    val animation = object : Animation() {
+        override fun applyTransformation(interpolatedTime: Float, t: Transformation?) {
+            layoutParams.height = currentHeight + ((targetHeight - currentHeight) * interpolatedTime).toInt()
+            requestLayout()
+        }
+
+        override fun willChangeBounds(): Boolean {
+            return true
+        }
+    }
+    animation.duration = 300
+    startAnimation(animation)
+}
+
+fun TextView.collapseTextView(lines: Int) {
+    val initialHeight = measuredHeight
+    setLines(lines)
+    val viewParent = parent
+    if (viewParent is View) {
+        val matchParentMeasureSpec = View.MeasureSpec.makeMeasureSpec(viewParent.width, View.MeasureSpec.EXACTLY)
+        val wrapContentMeasureSpec = View.MeasureSpec.makeMeasureSpec(0, View.MeasureSpec.UNSPECIFIED)
+        measure(matchParentMeasureSpec, wrapContentMeasureSpec)
+    }
+    val targetHeight = measuredHeight
+    minLines = 0
+    maxLines = Int.MAX_VALUE
+
+    val animation = object : Animation() {
+        override fun applyTransformation(interpolatedTime: Float, t: Transformation?) {
+            if (interpolatedTime == 1F) {
+                setLines(lines)
+            } else {
+                layoutParams.height = initialHeight - ((initialHeight - targetHeight) * interpolatedTime).toInt()
+                requestLayout()
+            }
+        }
+
+        override fun willChangeBounds(): Boolean {
+            return true
+        }
+    }
+
+    animation.duration = 300
+    startAnimation(animation)
+}
```

Comment:
I find it unnecessary to write 100 lines of code to do something which is just an enhancement and unrelated to the core working of the project. It can impact performance, and contain bugs, we have to maintain it. 2 months later, nobody will understand the code and will not be able to maintain it. Same thing happened in server where instead of using a well-maintained library, they wrote celery integration by hand. New version of celery came and it broke. If they were using a library, it would have been a matter of bumping the version. That is why we use libraries